src/api/ZoweExplorerApi.ts

```diff
@@ -0,0 +1,472 @@
+         * @returns {imperative.Session} a Zowe CLI Session
+         */
+        getSession(profile?: imperative.IProfileLoaded): imperative.Session;
+    }
+
+    /**
+     * API for providing a USS API handler to the extension.
+     * @export
+     */
+    export interface IUss extends ICommon {
+        /**
+         * Return the directory elements for a given USS path.
+         *
+         * @param {string} path
+         * @returns {IZosFilesResponse}
+         *     A response structure that contains a boolean success property
+         *     as well as the list of results in apiResponse.items with
+         *     minimal properties name, mode.
+         */
+        fileList(
+            path: string
+        ): Promise<zowe.IZosFilesResponse>;
+
+        /**
+         * Check th USS chtag.
+         *
+         * @param {string} USSFileName
+         * @returns {Promise<boolean>}
+         */
+        isFileTagBinOrAscii(
+            USSFileName: string
+        ): Promise<boolean>;
+
+        /**
+         * Retrieve the contents of a USS file.
+         * @param {string} ussFileName
+         * @param {zowe.IDownloadOptions} options
+         */
+        getContents(
+            ussFileName: string,
+            options: zowe.IDownloadOptions
+        ): Promise<zowe.IZosFilesResponse> ;
+
+        /**
+         * Uploads the files at the given path. Use for Save.
+         *
+         * @param {string} inputFile
+         * @param {string} ussname
+         * @param {boolean} [binary]
+         * @param {string} [localEncoding]
+         * @returns {Promise<zowe.IZosFilesResponse>}
+         */
+        putContents(
+            inputFile: string,
+            ussname: string,
+            binary?: boolean,
+            localEncoding?: string,
+            etag?: string,
+            returnEtag?: boolean
+        ): Promise<zowe.IZosFilesResponse>;
+
+        /**
+         * Uploads directory at the given path.
+         *
+         * @param {string} inputDirectory
+         * @param {string} ussname
+         * @param {IUploadOptions} [options]
+         * @returns {Promise<zowe.IZosFilesResponse>}
+         */
+        uploadDirectory(
+            inputDirectory: string,
+            ussname: string,
+            options: zowe.IUploadOptions
+        ): Promise<zowe.IZosFilesResponse>;
+
+        /**
+         * Create a new directory or file in the specified path.
+         *
+         * @param {string} ussPath
+         * @param {string} type
+         * @param {string} [mode]
+         * @returns {Promise<string>}
+         */
+        create(
+            ussPath: string,
+            type: string,
+            mode?: string
+        ): Promise<string>;
+
+        /**
+         * Deletes the USS file at the given path.
+         *
+         * @param {string} fileName
+         * @param {boolean} [recursive]
+         * @returns {Promise<zowe.IZosFilesResponse>}
+         */
+        delete(
+            fileName: string,
```

Comment:
I hope I'm not disrupting too much here.
I was just thinking maybe some consistency between path vs name would be nice.
I'm not really requesting for this to be changed at this time though.

Reply:
Done.